ugs-core/src/com/willwinder/universalgcodesender/utils/MathUtils.java

```diff
@@ -0,0 +1,123 @@
+/*
+    Copyright 2020 Will Winder
+
+    This file is part of Universal Gcode Sender (UGS).
+
+    UGS is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    UGS is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with UGS.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.willwinder.universalgcodesender.utils;
+
+import com.willwinder.universalgcodesender.model.PartialPosition;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A collection of common mathematical operations
+ *
+ * @author Joacim Breiler
+ */
+public class MathUtils {
+
+    public static final int COUNTER_CLOCKWISE = -1;
+    public static final int CLOCKWISE = 1;
+    public static final int COLLINEAR = 0;
+
+    /**
+     * To find orientation of ordered triplet (p1, p2, p3).
+     * https://www.geeksforgeeks.org/orientation-3-ordered-points/
+     *
+     * @param p1 a point in a polygon
+     * @param p2 a point in a polygon
+     * @param p3 a point in a polygon
+     * @return The function returns if the sequence of the points are COLINEAR, CLOCKWISE or COUNTER_CLOCKWISE
+     */
+    public static int orientation(PartialPosition p1, PartialPosition p2, PartialPosition p3) {
+        double value = (p2.getY() - p1.getY()) * (p3.getX() - p2.getX()) -
```

Comment:
Does this ever need to take the arc plane mode into consideration? Not having it be XY would by unusual but maybe there are cases where it matters?

Reply:
No it doesn't. I'm not exactly sure how this is supposed to work. Do you have any ideas?